components/PlantForm.js

```diff
@@ -0,0 +1,183 @@
+import styled, { css } from "styled-components";
+import { StyledButton } from "./StyledButton";
+
+const StyledFormWrapper = styled.form`
+  display: flex;
+  flex-direction: column;
+
+  ${css`
+    input[type="text"] {
+      width: 100%;
+      padding: 15px;
+      border-style: none;
+      border-radius: 10px;
+    }
+    textarea {
+      width: 100%;
+      padding: 15px;
+      border-style: none;
+      border-radius: 10px;
+      font-family: Arial, Helvetica, sans-serif;
+    }
+    h3 {
+      margin-bottom: 10px;
+    }
+    section {
+      display: flex;
+      flex-direction: row;
+      justify-content: start;
+    }
+    label {
+      /* FYI: genaue Anpassung in einer späteren User-Story (nach Wahl der Schriftart, Größe, etc.) */
+      margin-right: 15px;
+    }
+
+    input {
+      margin: 0 5px 0 0;
+    }
+
+    div.button {
+      margin-top: 15px;
+      display: flex;
+      justify-content: center;
+    }
+  `}
+`;
+
+export default function PlantForm({ onCreatePlant }) {
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    const selectedSeasons = formData.getAll("fertiliserSeason");
+
+    if (selectedSeasons.length === 0) {
+      // FYI: Hinzufügen einer universellen Error-Message für alle Pflichtfelder folgt
+      alert("Please select at least one season.");
+      return;
+    }
+    data.fertiliserSeason = selectedSeasons;
+    onCreatePlant(data);
+
+    event.target.reset();
+  }
+
+  const lightOptions = [
+    { id: "lightNeed1", value: "full sun", label: "Full Sun" },
+    { id: "lightNeed2", value: "partial shade", label: "Partial Shade" },
+    { id: "lightNeed3", value: "full shade", label: "Full Shade" },
+  ];
+
+  const waterOptions = [
+    { id: "waterNeed1", value: "low", label: "Low" },
+    { id: "waterNeed2", value: "medium", label: "Medium" },
+    { id: "waterNeed3", value: "high", label: "High" },
+  ];
+
+  const fertiliserOptions = [
+    { id: "fertiliserSeason1", value: "summer", label: "Summer" },
+    { id: "fertiliserSeason2", value: "spring", label: "Spring" },
+    { id: "fertiliserSeason3", value: "autumn", label: "Autumn" },
+    { id: "fertiliserSeason4", value: "winter", label: "Winter" },
+  ];
```

Comment:
Moving the `lightOptions`, `waterOptions`, and `fertiliserOptions` arrays outside of the `PlantForm` component.

💡 Defining these arrays inside the component function causes them to be recreated each time the component re-renders, which could lead to unnecessary memory usage. Moving them outside of the component ensures that they are defined only once, reducing memory overhead and improving performance. 

Example:

``` js

// all imports 

// move the arrays outside of the component.

  const lightOptions = [
    { id: "lightNeed1", value: "full sun", label: "Full Sun" },
    { id: "lightNeed2", value: "partial shade", label: "Partial Shade" },
    { id: "lightNeed3", value: "full shade", label: "Full Shade" },
  ];

  const waterOptions = [
    { id: "waterNeed1", value: "low", label: "Low" },
    { id: "waterNeed2", value: "medium", label: "Medium" },
    { id: "waterNeed3", value: "high", label: "High" },
  ];

  const fertiliserOptions = [
    { id: "fertiliserSeason1", value: "summer", label: "Summer" },
    { id: "fertiliserSeason2", value: "spring", label: "Spring" },
    { id: "fertiliserSeason3", value: "autumn", label: "Autumn" },
    { id: "fertiliserSeason4", value: "winter", label: "Winter" },
  ];

export default function PlantForm({ onCreatePlant }) { 
// other code 

```

Reply:
thanks